### ISSUE TITLE
bug(ipeps): _eval_fresh_2site post-convergence re-eval diverges from in-loop CTM at chi=8

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#317]
## Summary

After 2-site shared-tensor C4v AD optimization converges in-loop to the correct physical minimum, the post-convergence \`_eval_fresh_2site\` re-evaluation in \`src/tenax/algorithms/ipeps_optimize.py\` computes a different (worse) energy — typically off by ~1e-2 — from a fully-converged fresh-CTM run on the same A tensor.

Discovered while validating plan #315 / PR for \`feat/lorentzian-projector-backward\` (Approach A).

## Evidence

End-to-end benchmark, Heisenberg spin-1/2 AFM, D=2, χ=8, 2-site shared-tensor C4v, L-BFGS + Hager-Zhang + metric preconditioning, explicit-AD with Lorentzian projector backward:

\`\`\`
converged at step 32 (dE=8.806e-09 < tol=1.000e-08)
E_best (in-loop)        = -0.6602310494
final E (re-eval)       = -0.65001056    <-- off by ~0.01
literature (Corboz)     = -0.6548
QMC exact (Sandvik)     = -0.66944
\`\`\`

The in-loop \`E_best = -0.6602310494\` matches the hacked-swap diagnostic run at the same χ (\`-0.6602309466\`, issue #299 validation) to 7 digits, so the in-loop forward/backward chain is correct. The discrepancy is entirely in the re-eval step.

## Reproducer

\`bench_2site_c4v_diagnostic.py\` (untracked in this worktree) runs the full config at D=2, χ=8 on a 2×CUDA host and produces the mismatch. Any 2-site shared-tensor C4v Heisenberg config with \`gs_explicit_ad=True\`, \`projector_method="eigh"\`, and tight in-loop convergence reproduces it.

## Root-cause hypotheses (not yet narrowed down)

1. **Warm-start vs fresh-start divergence.** The in-loop CTM is warm-started from the previous AD step's environment and runs with \`max_iter=100, min_iter=30\`. The re-eval (\`_eval_fresh_2site\` → \`_ctm_tensor_multisite_fixed_point\`) runs from a random environment initialization with \`max_iter=200, conv_tol=1e-10\`. At χ=8 the CTM has multiple near-minima and different starting points or tolerances can land on different fixed points.

2. **Eigenvector sign freedom across different projector backward calls.** The plumbed \`truncated_eigh_regularized(-rho, k)\` and the baseline \`regularized_eigh(rho)[:,-k:][:,::-1]\` both return top-k eigenvectors of ρ in descending eigenvalue order, but the per-column sign can differ between \`jnp.linalg.eigh(-rho)\` and \`jnp.linalg.eigh(rho)\`. If the forward CTM is not perfectly invariant under per-column projector sign flips, this manifests as a small fixed-point drift. (Phase gauge forward_gauge="phase" is supposed to normalize this, but the degree of invariance should be audited.)

3. **Pre-existing drift independent of the new backward.** Before any Lorentzian plumbing, the hacked-swap diagnostic run produced \`final E = -0.6602161072\` vs in-loop \`-0.6602309466\` — a discrepancy of ~1.5e-5, three orders of magnitude smaller than the plumbed version's ~1e-2. So something *changed* between the two runs beyond just the code hoisting, but the in-loop convergence itself is bit-for-bit identical at E_best. The difference must be in the sequence of converged A tensors fed into the re-eval.

## Impact

- **In-loop energies are correct**, so the publishable benchmark is the \`E_best = -0.66023\` number from the in-loop trajectory, not \`final E\`.
- Any user reading \`optimize_gs_ad\`'s return value will see the misleading -0.65 final energy. Tests asserting on the return value will be brittle.
- The \`_eval_fresh_2site\` path has long claimed to "compare fully-converged fresh evaluations" as a safety net; in the current state it is less trustworthy than the in-loop \`E_best\` — an inversion of the original design intent.

## Relationship to plan PR #315

The plan's Task 12 slow-tier benchmark should assert on the **in-loop best** energy, not the \`optimize_gs_ad\` return value, until this is resolved. Otherwise the test will fail even though the feature works.

## Suggested investigation steps

1. Add a \`_ctm_tensor_multisite_fixed_point\` warm-start test: re-run the fresh CTM starting from the final in-loop environment instead of a random init, and compare against the in-loop energy. If they match, the drift is from cold-start random init.
2. Audit \`_phase_fix_ctm_tensor\` and \`_gauge_fix_ctm_tensor\` for per-column sign invariance.
3. Run \`_eval_fresh_2site\` with increasing \`conv_tol\` (1e-6, 1e-8, 1e-10, 1e-12) and \`max_iter\` (100, 200, 500) and plot the final E to see whether the drift is monotone (stuck in a nearby minimum) or oscillating (unstable fixed point).
4. Audit history: \`git log -p src/tenax/algorithms/ipeps_optimize.py\` around \`_eval_fresh_2site\` for changes that may have subtly altered which fields land in \`eval_config2\`.

cc @tenax-lab
